mpt/impl/smtp/cassandra-rabbitmq-object-storage/src/test/java/org/apache/james/mpt/smtp/CassandraRabbitMQAwsS3SmtpTestRule.java

```diff
@@ -0,0 +1,76 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mpt.smtp;
+
+import org.apache.commons.configuration.DefaultConfigurationBuilder;
+import org.apache.james.CassandraJamesServerMain;
+import org.apache.james.CleanupTasksPerformer;
+import org.apache.james.GuiceJamesServer;
+import org.apache.james.backend.rabbitmq.DockerRabbitMQSingleton;
+import org.apache.james.backends.cassandra.init.configuration.ClusterConfiguration;
+import org.apache.james.dnsservice.api.DNSService;
+import org.apache.james.modules.TestAwsS3BlobStoreModule;
+import org.apache.james.modules.TestRabbitMQModule;
+import org.apache.james.modules.blobstore.BlobStoreChoosingModule;
+import org.apache.james.modules.protocols.SmtpGuiceProbe.SmtpServerConnectedType;
+import org.apache.james.modules.rabbitmq.RabbitMQModule;
+import org.apache.james.modules.server.CamelMailetContainerModule;
+import org.apache.james.queue.api.MailQueueItemDecoratorFactory;
+import org.apache.james.queue.api.RawMailQueueItemDecoratorFactory;
+import org.apache.james.server.core.configuration.Configuration;
+import org.apache.james.util.Host;
+
+import com.google.inject.util.Modules;
+
+public class CassandraRabbitMQAwsS3SmtpTestRule extends SmtpTestRule {
+
+    public CassandraRabbitMQAwsS3SmtpTestRule(SmtpServerConnectedType smtpServerConnectedType, Host cassandraHost) {
+        super(cassandraHost, smtpServerConnectedType);
+    }
+
+    @Override
+    protected GuiceJamesServer createJamesServer() throws Exception {
+        Configuration configuration = Configuration.builder()
+                .workingDirectory(folder.newFolder())
+                .configurationFromClasspath()
+                .build();
+
+        return GuiceJamesServer.forConfiguration(configuration)
```

Comment:
this setup is messy: you should be able to combine/override everything with GuiceJamesServer API, at least it would decrease the level on nesting.
And please choose the same style for everything: either vararg method or single entry one

Reply:
I do not understand, maybe we have to pair on that.